functions/savePendingContacts.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+import '@twilio-labs/serverless-runtime-types';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
+import { responseWithCors, bindResolve, error500, success } from '@tech-matters/serverless-helpers';
+
+const TokenValidator = require('twilio-flex-token-validator').functionValidator;
+
+type EnvVars = {
+  SYNC_SERVICE_SID: string;
+  SAVE_CONTACT_FN: string;
+};
+
+type Nullable<T> = T | undefined | null;
+type SaveContactFn = (payload: any) => Promise<any>;
+
+/**
+ * Gets a promisefied saveContactFn() given a function name.
+ *
+ * @param functionName Name of the Serverless Function (e.g. 'saveContactToSaferNet')
+ * @param context Context
+ * @param event Event
+ * @returns @type {Nullable<SaveContactFn>} saveContactFn
+ */
+const getSaveContactFn = (
+  functionName: string,
+  context: Context,
+  event: any,
+): Nullable<SaveContactFn> => {
+  const functionPath = Runtime.getFunctions()[functionName]?.path;
+  if (!functionPath) return;
+
+  const saveContactHandler = require(functionPath)?.handler;
+  if (!saveContactHandler) return;
+
+  // eslint-disable-next-line consistent-return
+  return payload =>
+    new Promise((resolveCallback, rejectCallback) => {
+      // Callback passed to saveContactHandler
+      const callback: ServerlessCallback = (error: any, callbackPayload: any) => {
+        const isError = error || ![200, 204].includes(callbackPayload?.statusCode);
+        return isError ? rejectCallback(callbackPayload) : resolveCallback(callbackPayload);
```

Comment:
Would we not need to `rejectCallback(error)` if error was set?

Reply:
Good catch! (although we're not using whatever comes back)

Reply:
Fixed